### PR TITLE
Deepspeed evoformer attention

### PR DESCRIPTION
**Pull request by christinaflo**

### CODE REVIEW

File: tests/test_deepspeed_evo_attention.py [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+        q_repro = clone(q)
+        kv_repro = clone(kv)
+        biases_repro = [clone(b) for b in biases]
+
+        a = init_attn()
+        out_repro = a(q_repro, kv_repro, biases=biases_repro, use_deepspeed_evo_attention=True)
+        loss_repro = torch.mean(out_repro)
+        loss_repro.backward()
+
+        q_gt = clone(q)
```
Comment (jnwei) [inline tests/test_deepspeed_evo_attention.py:131]: What does *_gt mean here?

File: tests/test_deepspeed_evo_attention.py [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+    def test_ds_kernel_vs_attention_forward(self):
+        """Compare regular attention vs. DeepSpeed Evoformer kernel."""
+        self.compare_attention_types(use_flash=False)
+
+    @compare_utils.skip_unless_flash_attn_installed()
+    def test_ds_kernel_vs_flash_attn_forward(self):
+        """Compare Flash Attention vs. DeepSpeed Evoformer kernel."""
+        self.compare_attention_types(use_flash=True)
+
+    def test_ds_kernel_vs_attention_backward(self):
```
Comment (jnwei) [inline tests/test_deepspeed_evo_attention.py:84]: Could you please write a few comments to help explain the comments of this test?

File: README.md [hunk 3]
```diff
@@ -39,13 +39,14 @@ kernels support in-place attention during inference and training. They use
 implementations, respectively.
 - **Efficient alignment scripts** using the original AlphaFold HHblits/JackHMMER pipeline or [ColabFold](https://github.com/sokrypton/ColabFold)'s, which uses the faster MMseqs2 instead. We've used them to generate millions of alignments.
 - **FlashAttention** support greatly speeds up MSA attention.
+- **DeepSpeed DS4Sci_EvoformerAttention kernel** is a memory-efficient attention kernel developed as part of a new collaboration between OpenFold and DeepSpeed4Science initiative. The kernel provides substantial speedups for training and inference, and significantly reduces the model's peak device memory requirement by 13X. The model is 15% faster during the initial training and finetuning stages, with an overall of 40% lower peak memory consumption. To use this feature, simply set the `use_deepspeed_evo_attention` option in `openfold/config.py`.
```
Comment (jnwei) [inline README.md:42]: Suggested re-phrasing for the description:

**DeepSpeed DS4Sci_EvoformerAttention kernel** is a memory-efficient attention kernel developed as part of a collaboration between OpenFold and the DeepSpeed4Science initiative.

**Review by jnwei — APPROVED** [review]
Overall looks good, thanks!

Just a few very minor comments.

File: tests/test_deepspeed_evo_attention.py (hunk 1 above)
Comment (christinaflo) [reply tests/test_deepspeed_evo_attention.py:131]: ground truth

**Pull request merged**